app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt

```diff
@@ -152,7 +152,9 @@ class LoginFragment : Fragment() {
 
     override fun onStart() {
         super.onStart()
-        smartAuthViewModel.requestCredentials(activity)
+        if (!smartAuthViewModel.requestedCredentialsEarlier) {
+            smartAuthViewModel.requestCredentials(activity)
+        }
```

Comment:
Why are you exposing and accessing the variable from activity. Make the `requestedCredentialsEarlier` private

Reply:
The current method name of the request credentials method is "requestCredentials". I thought that the name implies that calling it will always request credentials by showing a popup. So, to make the fact that we only request credentials once more explicit, I exposed the variable public and used it in the Activity.

Can you please tell me what should I do?

Reply:
Return from the viewmodel if it is requested more than once, why give control to the fragment?

Reply:
I'll push the changes right away